apps/aecore/src/aec_dev_reward.erl

```diff
@@ -0,0 +1,122 @@
+-module(aec_dev_reward).
+
+-export([ensure_env/0]). % setup hook
+
+-export([enabled/0,
+         activated/1,
+         beneficiaries/0,
+         allocated_shares/0,
+         total_shares/0,
+         split/3]).
+
+-include_lib("aecontract/include/hard_forks.hrl").
+
+
+-define(ENABLED, false).
+-define(TOTAL_SHARES, 1000). % 100 shares == 10% of the reward
+%%% for: "ak_2uN1CEVGs4D5QnFudhp8dSqrwJEMFiTfwvbCK3SJzVYRDiusfd:100" TODO - use actual pubkey for real account
+-define(BENEFICIARIES, [{<<250,151,56,184,99,123,38,230,217,93,156,146,231,90,209,233,15,203,25,102,32,140,178,207,59,171,81,2,249,197,198,61>>, 100}]).
+
+ensure_env() ->
+    Enabled = cfg(<<"protocol_beneficiaries_enabled">>, ?ENABLED),
+    Benefs0 = cfg(<<"protocol_beneficiaries">>, lists:map(fun unparse_beneficiary/1, ?BENEFICIARIES)),
+    case Enabled andalso parse_beneficiaries(Benefs0) of
+        false ->
+            application:set_env(aecore, dev_reward_enabled, false);
+        {ok, Beneficiaries} ->
+            {_, Shares} = lists:unzip(Beneficiaries),
+            AllocShares = lists:sum(Shares),
+            case AllocShares =< ?TOTAL_SHARES of
```

Comment:
Any check on AllocShares positive?

Reply:
The shares forming the sum of shares must pass the regex in config which prevents negative numbers.

Reply:
I see check on non negative. I am asking on check on AllocShares positive (non-zero) though.

Reply:
I am adding a if-branch for this case so it fails if it's 0.